arrays/arrays.c

```diff
@@ -83,14 +107,34 @@ char *arr_read(Array *arr, int index) {
 void arr_insert(Array *arr, char *element, int index) {
 
   // Throw an error if the index is greater than the current count
+  if (index > arr->count) {
+    switch(errno) {
+      default:
+        fprintf(stderr, "Inserting into array -> Error: Index is out of bounds \n");
+    }
+
+    exit(1);
+  }
 
   // Resize the array if the number of elements is over capacity
+  if (arr->count + 1 > arr->capacity) {
+    resize_array(arr);
+  }
 
   // Move every element after the insert index to the right one position
+  for (int i = index; i < arr->count; i++) {
+    // Incrementing this to i + 2 gives the correct result but failed tests,
+    // but after some debugging and doing `make tests` i + 1 returns
+    // all tests passed, but somehow display incorrect elements in array.
+    arr->elements[i+1] = arr->elements[i];
+  }
 
   // Copy the element and add it to the array
+  char* elem = strdup(element); // Also adding this line here and addign that to the arr->elements[index] causes a segmentation fault.
+  arr->elements[index] = element;
```

Comment:
In regards to your comment about the segmentation fault that's probably to do with an "off-by-one" type error. I can't say exactly where without testing the code more carefully but probably in one of your for loops or manipulations of index. I ran into the same problem when I worked through this module yesterday.